Exercises/NoModules/Chapter 16/Soln16_03/Truckload.cpp

```diff
@@ -0,0 +1,146 @@
+#include "Truckload.h"
+
+#include <iostream>
+#include <stdexcept>    // For standard exception type std::out_of_range 
+#include <string>       // For std::string and std::to_string()
+
+// Definition of the nested class member
+// Since this member is private, 
+// its definition can be moved to the source file.
+class Truckload::Package
+{
+public:
+  SharedBox m_box;      // Pointer to the Box object contained in this Package
+  Package* m_next;      // Pointer to the next Package in the list
+
+  Package(SharedBox box) : m_box{ box }, m_next{ nullptr } {} // Constructor
+  ~Package() { delete m_next; }                           // Destructor
+};
+
+// Constructor - one Box (moved to source file to gain access to definition of Package)
+Truckload::Truckload(SharedBox box)          
+{
+  m_head = m_tail = new Package{ box };
+}
+
+// Constructor - vector of Boxes
+Truckload::Truckload(const std::vector<SharedBox>& boxes)
+{
+  for (const auto& box : boxes)
+  {
+    addBox(box);
+  }
+}
+
+// Copy constructor
+Truckload::Truckload(const Truckload& src)
+{
+  for (Package* package{ src.m_head }; package; package = package->m_next)
+  {
+    addBox(package->m_box);
+  }
+}
+
+Truckload& Truckload::operator=(const Truckload& src)
+{
+  // Delete and reset the current list
+  delete m_head;
+  m_head = m_tail = nullptr;
+
+  // Add all boxes from the src truckload
+  for (Package* package{ src.m_head }; package; package = package->m_next)
+  {
+    addBox(package->m_box);
+  }
+}
```

Comment:
operator= must return a value.

Reply:
Since this method was wrong, please verify that the rest of this implementation matches with what's in chapter 11.